src/validation.cpp

```diff
@@ -2205,6 +2210,34 @@ static void AppendWarning(bilingual_str& res, const bilingual_str& warn)
 
 void CChainState::UpdateTip(const CBlockIndex* pindexNew)
 {
+    auto& func_name = __func__;
+    auto log_progress = [this, pindexNew, &func_name](
```

Comment:
1142b4f Seeing this warning like @Sjors reported. What CI failure were you seeing when it was addressed?
```
validation.cpp:2203:44: warning: lambda capture 'func_name' is not required to be captured for this use [-Wunused-lambda-capture]
    auto log_progress = [this, pindexNew, &func_name](
                                        ~~~^~~~~~~~~
1 warning generated.
```

Reply:
How did you address it when the CI didn't pass? e.g. with `[this, pindexNew]`, or `[&]`, or using a struct instead of a lambda?

Reply:
Can't dig up the exact stacktrace, but on one of the more exotic CI platforms (mac, win, or mingw32), the variable isn't captured unless explicitly named, and compilation fails. I think I tried your first two suggestions, but am not sure what you mean by "struct instead of a lambda."

Reply:
> Can't dig up the exact stacktrace, but on one of the more exotic CI platforms (mac, win, or mingw32), the variable isn't captured unless explicitly named, and compilation fails. I think I tried your first two suggestions, but am not sure what you mean by "struct instead of a lambda."

Related: I did have an earlier suggestion for eliminating the lambda here: https://github.com/bitcoin/bitcoin/pull/21526#discussion_r672599645. Instead of using the lambda to repeat the logging at the beginning and end of the function, just log without a lambda at the beginning of the function

I don't think is too important, though

Reply:
@ryanofsky good point - I'll give that a shake, and take a look at addressing your other feedback while I'm at it.

Reply:
This is new to me, but if I understand correctly, the issue is compiler/system-dependent lambda capture of `__func__`. Maybe convert the `log_progress` lambda to a function: https://github.com/jonatack/bitcoin/commit/e95e25e8aa6581d8658b05581f0b1a04e8712387.

Reply:
(by "struct instead of a lambda", I was thinking of a struct like [`38a81a8` (#21261)](https://github.com/bitcoin/bitcoin/pull/21261/commits/38a81a8e20b0e5ad9fef0eae8abd914619f05b25) that I originally wrote as a lambda, before turning it into a struct to be more similar to the other peer eviction comparators)

Reply:
Took @jonatack's suggestion and moved the lambda to a static function.